File: mipush_hook/src/main/java/com/nihility/OuterDependencies.java
package com.nihility;

import com.nihility.service.XMPushServiceListener;
import com.xiaomi.push.service.XMPushService;

public interface OuterDependencies {
    Configurations configuration();

    XMPushServiceListener serviceListener(XMPushService pushService);

    HookedMethodHandler hookedMethodHandler();
}

File: mipush_hook/src/main/java/com/nihility/OuterDependencies.kt
package com.nihility

import com.nihility.service.XMPushServiceListener
import com.xiaomi.push.service.XMPushService

interface OuterDependencies {
    fun configuration(): Configurations?
    fun serviceListener(pushService: XMPushService): XMPushServiceListener?
    fun hookedMethodHandler(): HookedMethodHandler?
}
